.github/workflows/publish.yml

```diff
@@ -0,0 +1,45 @@
+name: Publish Package
+on:
+    push:
+        branches:
+            - main
+        paths:
+            - 'package.json'
+jobs:
+    publish:
+        runs-on: ubuntu-latest
+        permissions:
+            contents: read
+            packages: write
+        steps:
+            - name: Checkout repository
+              uses: actions/checkout@v2
+
+            - name: Check if version has been updated
+              id: check
+              uses: EndBug/version-check@v1
+              with:
+                  diff-search: true
+                  token: ${{ secrets.GITHUB_TOKEN }}
+
+            - name: Version update detected
+              if: steps.check.outputs.changed == 'true'
+              run: 'echo "Version change found! New version: ${{ steps.check.outputs.version }} (${{ steps.check.outputs.type }})"'
+
+            - name: Setup node
+              if: steps.check.outputs.changed == 'true'
+              uses: actions/setup-node@v2
+              with:
+                  node-version: '16.x'
+                  registry-url: 'https://npm.pkg.github.com'
+                  scope: '@faceteer'
+
+            - name: Installing dependencies
+              if: steps.check.outputs.changed == 'true'
+              run: npm ci
+
+            - name: Publishing package
+              if: steps.check.outputs.changed == 'true'
+              run: npm publish
+              env:
+                  NODE_AUTH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```

Comment:
I don't believe this token has access to publish to the NPM registry. Only Github.

Reply:
@s-fletcher I think one drawback here is also that you still need to determine what type of release it will be, and you need to keep the package.json version up to date in every PR. Might not be a problem in this repo, but if you have more than 1 PR open, you will always have conflicting version numbers on PR merges.

> > What's the benefit of this over something like `semantic-release`?
> 
> @mckalexee For this solution, you're not required to format commit messages a certain way. All it does is see that the version in `package.json` has changed.
> 
> I personally believe the approach in this PR is more straight forward and simpler, yet it lacks additional features that `semantic-release` provides (generating release notes, notifying maintainers, etc).